sigal/image.py

```diff
@@ -268,6 +272,43 @@ def get_iptc_data(filename):
     return iptc_data
 
 
+def get_image_metadata(filename):
+    logger = logging.getLogger(__name__)
+    try:
+        img = _read_image(filename)
+    except Exception as e:
+        logger.error('Could not open image %s metadata: %s',
+                     filename, e)
+
+    exif = None
+    iptc = None
+    size = None
+
+    try:
+        exif = get_exif_data(img)
+    except Exception as e:
+        logger.warning('Could not read EXIF data from %s: %s',
+                       filename, e)
+
+    try:
+        iptc = get_iptc_data(img)
+    except Exception as e:
+        logger.warning('Could not read IPTC data from %s: %s',
+                       filename, e)
+
+    try:
+        size = get_size(img)
```

Comment:
It seems that size is not used later ?

Reply:
Size is not used for the source image. But it used for other images, such as thumbnails.
My goal was to code a function that extract everything that could be needed. And `get_size()` with an already opened image is really harmless from a performance point-of-view:
![Performance with snakeviz](https://screenshotscdn.firefoxusercontent.com/images/37a29adb-6da1-49da-bb91-17baca784c2d.png)
(Notice how calling `Image.open` will call `JpegImagePlugin._open` twice here).